src/System.ComponentModel.TypeConverter/src/System/ComponentModel/InterlockedBitVector32.cs

```diff
@@ -0,0 +1,59 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Threading;
+
+namespace System.ComponentModel
+{
+    /// <summary>
+    /// Provides a subset of the <see cref="BitVector32"/> surface area, using volatile operations for reads and interlocked operations for writes. 
+    /// </summary>
+    internal struct InterlockedBitVector32
+    {
+        private int _data;
+
+        public InterlockedBitVector32(int data)
+        {
+            _data = data;
+        }
+
+        public bool this[int bit]
+        {
+            get { return (Volatile.Read(ref _data) & bit) == bit; }
+            set
+            {
+                while (true)
+                {
+                    int oldValue = _data;
+                    int newValue = value ? oldValue | bit : oldValue &= ~bit;
```

Comment:
Not in the current code.  And removing that would shave off a tiny bit of IL and a (very predictable) branch.  But changing that would end up impacting the exposed API from this helper, which I'd like to keep as close to BitVector32 as possible, in order to keep this code as close to the same desktop code as possible, and I highly doubt it would lead to any measurable performance impact, given that we're using reflection, interlocked APIs, etc.  I'd like to keep it as-is unless it's shown to actually be impactful.